Adds the ability to add a VCPU to a VM

benlemasurier opened the pull request.

benlemasurier committed:
Adds the ability to add a VCPU to a VM

mdlayher commented on darity.go:
```diff
@@ -51,6 +54,9 @@ var (
 	ErrIncorrectVersion = errors.New("incorrect KVM version")
+
+	// ErrTooManyVCPUS is returned when a more than kvmCapMaxCPUS is requested.
+	ErrTooManyVCPUS = fmt.Errorf("a maximum of %d VCPUs are supported.", kvmCapNrVCPUS)
```
There shouldn't be a period on this error message.

mdlayher commented on darity.go:
```diff
@@ -51,6 +54,9 @@ var (
 	//   value other than 12.
 	ErrIncorrectVersion = errors.New("incorrect KVM version")
+
+	// ErrTooManyVCPUS is returned when a more than kvmCapMaxCPUS is requested.
```
Comment should mention `kvmCapNrVCPUS` instead?

mdlayher commented on darity.go:
```diff
@@ -228,6 +237,22 @@ func (v *VM) AddMemorySlot(n uint64, flags MemorySlotFlag) error {
 }
 
+// AddVCPU adds n VCPUs to a virtual machine.
+func (v *VM) AddVCPU(n uint64) error {
```
This could probably just be a regular `int`, but that would mean also having to verify that values of less than 0 are invalid.  Maybe just `uint` or `uint8` instead?  `uint64` is overkill for the range 0-9.

benlemasurier committed:
cleanup

- use a tiny value where tiny values are needed. uint64 is a bit much
  for a cpu count.

- comment cleanup

mdlayher commented:
:+1: !
